messagecatalog/spec.md

```diff
@@ -0,0 +1,1059 @@
+respected.
+
+For centralized governance scenarios, the `authority` SHOULD be set to a URI
+reflecting the governing entity, like `https://messagecatalog.corp.example.com`.
+
+### 2.2.2. Message definition version URI
+
+To refer to a specific message definition object, unambiguous references to
+specific [definition versions](#22-schema-and-schema-version) are needed. These
+references can be resolved into definition documents with the help of a catalog.
+
+The message definition version URI is composed from the
+[authority](#authority-definition-authority) (as the base URI) and the
+[message definition version id](#id-definition-version-id).
+
+In the default case, this URI corresponds directly to the URI of the
+["getMessage definitionVersion"](#342-get-a-specific-schema-version) API
+operation, which is convenient for simple usage scenarios.
+
+In [replication scenarios](#4-replication-model-and-state-change-events), the
+URI might not correspond to a resolvable network address, as permitted for the
+authority URI in the previous section. A consumer SHOULD then be configured with
+a fixed message catalog endpoint for it to use, and obtain definitions
+identified by a URI using the
+["getMessage definitionVersionByURI"](#342-get-a-specific-schema-version) API
+operation on that endpoint, rather than trying to resolve the URI directly.
+
+The following table shows some examples for a catalog hosted at
+`https://example.com/messagecatalog`:
+
+| [authority](#authority-definition-authority) | [id](#id-definition-version-id)                               | Message definition URI                                                                         |
+| -------------------------------------------- | ------------------------------------------------------------- | ---------------------------------------------------------------------------------------------- |
+| (empty, therefore implied)                   | `342`                                                         | <https://example.com/messagecatalog/342>                                                       |
+| <https://example.com/messagecatalog>         | `342`                                                         | `https://example.com/messagecatelog/342`                                                       |
+| `http://messagecatalog.example.com/`         | `/mygroup/messagedefinitions/myMessage definition/versions/2` | <http://messagecatalog.example.com/mygroup/messagedefinitions/myMessage definition/versions/2> |
+| `http://messagecatalog.example.org/`         | `http://messagedefinitions.example.org/41751`                 | <http://messagecatalog.example.org/41751>                                                      |
+| `urn:example:messagecatalog`                 | `a1b2c3d4`                                                    | `urn:example:messagecatalog:a1b2c3d4`                                                          |
```

Comment:
The URN example might be tricky for a generic player, as to my understanding the `urn` scheme is not hierarchical, if I understand [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986/#section-3.3) correctly.